docs/lifecycle-management.md

```diff
@@ -0,0 +1,185 @@
+# Higher level VM Management
+
+## Motivation
+
+So far KubeVirt provides a pod-like interface when managing VMs. This seems to
+be a good building block.
+On the other hand this might not be sufficient for usual virtualization
+management UIs, which do expect to manage the complete life-cycle of a VM,
+including shut down VMs 8which is currently not intended by the core runtime).
+
+
+This document outlines how minimal assumptions can be met and also visits how
+other requirements impact the design.
+
+
+## Requirements
+* Running and non-running VMs are handled through the same API
+* Creating a VM must be separate from launching a VM
+* Support for “flavor” based VMs (templating)
+* Support for expanding sparse VM definitions
+* Support for detailed VM definitions
+* Support hot-plugging
+* Representation of dynamic instance specific data
+* Support for defaulting (to support OS compatibility)
+Considerations
+* Sub-resources for actions
+* Presets to add certain configurations
+
+
+## Definitions
+* **Defaulting** - Setting certain defaults during expansion (i.e. virtio for a
+  bus type, or e1000 for as a nic model)
+* **Expansion** - VM definitions can be provided in a sparse way i.e. just a
+  disk. Once the VM is getting started, the runtime will infer additional
+  requirements for this device and add them to the runtime VM definition.
+  This process of inferring and adding additional devices is called expansion.
+* **Flavor** - Comparable to a template or a specific instance type
+* **Preset** - Presetting certain values in a specified path of the document
+  (i.e. setting .spec.domain.devices.nvram). It is not about setting certain
+  default values for undefined fields (i.e. set virtio for all busses)
```

Comment:
So I'd actually move defaulting to be closer to the expansion. Because during the expansion it is decided what default values to use if nothing is provided.

To some degree we could say that Config is one or more presets. But in K8s presets are limited to certain paths in the document. In other words Presets all a partial override, not a complete spec override. However, we could adjust this to our needs as needed.

But this lead me to defaulting again - If me make the defaulting part of the spec, the presets could have value here.
With defaulting I'm referring to what default types/models are selected upon expansion of a spec. If the defaults become part of the spec, then the expander (who performs defaulting for me right now), could pull these values from the spec.
Finally, if the defaults are provided in such a way - through the spec - then presets could be used efficiently as well.

Example:
```yaml
Kind: VMPreset
metadata:
  name: SuSE_10_Defaults
spec:
  selector:
    matchLabels:
      distro: suse-10
  domain:
    default:
      devices:
        interfaces:
          model: virtio
…

Kind: VM
metadata:
  labels:
    distro: suse-10
spec:
  domain:
…
```

In that sense we could use presets to:
- Provide default templates for i.e. os compatability (i.e. set default)
- Provide building blocks to assemble VMs (i.e. snippet with a specific boot disk, efi support)

/cc @rmohr thoughts?